src/contact.rs

```diff
@@ -2708,7 +2708,18 @@ Hi."#;
 
         bob.recv_msg(&sent_msg).await;
         let contact = Contact::get_by_id(&bob, *contacts.first().unwrap()).await?;
-        assert!(contact.was_seen_recently());
+
+        let green = ansi_term::Color::Green.normal();
+        assert!(
+            contact.was_seen_recently(),
+            "{}",
+            green.paint(
+                "\nNOTE: This test failure is probably a false-positive, caused by tests running in parallel and shifting the time with `SystemTime::shift()`.
+Until the false-positive is fixed:
```

Comment:
> I think this node better goes to readme where there is a chance it will be read

but directly in the test output, there is a pretty large chance that it would be read there. i would have seen the hint (but i would not expect sth. about that in the README :)

my 2ct, feel free to ignore :)

Reply:
> Does not seem like it is going to be fixed

Hard to say, we'll see how annoying it turns out to be.

> Also if this is fixed, this assert will likely be forgotten about.

I would assume that if someone fixes it, it's because they saw the assert's error message, and then they will probably also remove the assert. If not, the assert isn't going to harm anyone until someone notices and removes it.

> so, cargo test cannot be used any longer? isn't that quite bad and worth targeting?

Idk how bad it is (I personally never run `cargo test`, always `cargo nextest run`, for which I created a shell alias), but it turned out to be kinda annoying to fix.

> for understanding: the issue is that the test-ultil shift() changes the time for all threads doing tests - and not only for the running test?

Yes; I'll add this explanation to the output for clarification.

> but directly in the test output, there is a pretty large chance that it would be read there. i would have seen the hint (but i would not expect sth. about that in the README :)

+1

---

In order not to block on what seems to me quite unimportant discussions, I'm going to merge in the hope that @link2xt's comment was not meant to be blocking. I'm happy to create or approve a follow-up PR.